packages/safe-apps-onboard/src/wrapper.ts

```diff
@@ -5,51 +5,56 @@ import { Initialization, API, Subscriptions, ConfigOptions, UserState } from 'bn
 
 export class OnboardWrapper implements API {
   private sdk = new SafeAppsSDK();
+  private triedToConnect = false;
   private onboardApi: API;
   private subscriptions?: Subscriptions;
   private safe: SafeInfo | undefined;
   private state: UserState | undefined;
+
   constructor(options: Initialization) {
     this.onboardApi = OnboardApi(options);
     this.subscriptions = options.subscriptions;
-    this.checkSafeApp().catch(console.log);
   }
 
-  async connectedSafe(timeout?: number): Promise<SafeInfo | undefined> {
-    if (!this.safe)
+  private async connectToSafe(timeout = 200): Promise<SafeInfo | undefined> {
```

Comment:
I increased the timeout because some developers complained they see onboard.js wallet selection modal. There's no evidence that timeout is the reason, but I increased it just in case.